### PR TITLE
karmada-search: support field selector for corev1 resources

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
karmada-search: support field selector for corev1 resources.

This supports most scenarios that use fieldSelector, such as finding a list of pods by node name.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
without this patch:
```sh
 # kubectl get pod --kubeconfig karmada-search.config --field-selector="spec.nodeName=member1-worker"
Error from server (BadRequest): Unable to find "/v1, Resource=pods" that match label selector "", field selector "spec.nodeName=member1-worker": "spec.nodeName" is not a known field selector: only "metadata.name", "metadata.namespace"
```

with this patch:
```sh
 # kubectl get pod --kubeconfig karmada-search.config --field-selector="spec.nodeName=member1-worker"
example-app-6c847589df-xmjm5                2024-10-29T03:15:25Z
...
```

**Does this PR introduce a user-facing change?**:

```release-note
karmada-search: support field selector for corev1 resources.
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Cool! Thanks~
/assign

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (XiShanYongYe-Chang) [inline pkg/search/proxy/framework/plugins/cache/scheme.go:30]: Resources that are not listed here can't be processed, can they? So how do you determine which resources can be listed here? In other words, are there any plans to add resources that are not listed?

Comment (SataQiu) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:30]: Thanks @XiShanYongYe-Chang I checked the resource kinds of Kubernetes and the corev1 group is the most important and the most common used. IMO, we should support the corev1 resource kinds first, and the other resource types can be gradually added based on community feedback.

Comment (XiShanYongYe-Chang) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:30]: @SataQiu, thanks for your quick reply!

Where does the Kubernetes community maintain Corev1 resources?

> Waiting for the Kubernetes runtime scheme package to support automatic ConversionFuncs registration, the community is trying to remove the legacyscheme

Is the Kuberenetes community working on this plan? Do we have to rely on this plan to add resources in the future?

Comment (XiShanYongYe-Chang) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:30]: Let me understand method two again, which means that when the upstream community implements this, other resources will automatically support it, and we don't need to register additional conversion functions. Am I right?

Comment (SataQiu) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:30]: Yes, you understand correctly.

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (XiShanYongYe-Chang) [inline pkg/search/proxy/framework/plugins/cache/scheme.go:122]: For these methods, can we directly import Kubernetes open functions?

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (XiShanYongYe-Chang) [inline pkg/search/proxy/framework/plugins/cache/scheme.go:135]: I don't seem to find any service code in the Kubernetes repository. Can you point it out?

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot~

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (XiShanYongYe-Chang) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:30]: Thanks~

> According to users feedback, support more resource types, such as apps, these can be added in subsequent PRs

I think we can do what you say. In addition, can you help create an issue to track this task, and if we don't get feedback on other resources for a long time, we may be able to proactively plan in the future.

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (SataQiu) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:122]: Emm... I guess we can't, because these functions are defined in the k/k repository, but we can't import it directly, which is not recommended.

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (SataQiu) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:135]: https://github.com/kubernetes/kubernetes/blob/6b031e50b2abfe15434beef6c7e01056091a26ea/pkg/apis/core/v1/conversion.go#L494-L507

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (XiShanYongYe-Chang) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:122]: Okay, I get it. It doesn't exit in the https://github.com/kubernetes/api repo.

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (XiShanYongYe-Chang) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:135]: Oh, I can see why I can't find it, it hasn't existed since version 1.31 and my code is still in v1.28.

File: pkg/search/proxy/framework/plugins/cache/scheme.go
Comment (XiShanYongYe-Chang) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:30]: Would it be better if we added a comment to the file stating that the implementation is referenced from xxx?

Comment (SataQiu) [reply pkg/search/proxy/framework/plugins/cache/scheme.go:30]: Good idea, updated.

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/lgtm
/cc @yanfeng1992 @RainbowMango

**Comment by yanfeng1992** [comment]
lgtm

**Review by yanfeng1992 — APPROVED** [review]

**Comment by SataQiu** [comment]
@XiShanYongYe-Chang need approve, PTAL!

**Comment by XiShanYongYe-Chang** [comment]
> @XiShanYongYe-Chang need approve, PTAL!

Let's wait for a review from @RainbowMango 

Hi @SataQiu, are you using Karmada now?

**Comment by RainbowMango** [comment]
/assign
But hoping the author @ikaven1024 could take a look.

**Comment by SataQiu** [comment]
ping @ikaven1024

**Pull request merged**
